crates/nu-protocol/src/errors/cli_error.rs

```diff
@@ -11,6 +16,9 @@ use miette::{
 };
 use thiserror::Error;
 
+/// The thread-global log of reported errors and warnings, so specific reports are only shown once
+static REPORT_LOG: LazyLock<Mutex<ReportLog>> = LazyLock::new(Mutex::default);
```

Comment:
I know I suggested the `LazyLock` but now I'm unsure if we should have it in the `EngineState` instead. We have access to it basically everywhere and this static doesn't work well, when you run with multiple engines, which is not relevant to us but maybe embedders.

Reply:
will be doing this as discussed in meeting